api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -155,6 +158,16 @@ class HttpTraceContext : public context::propagation::TextMapPropagator
       return SpanContext::GetInvalid();
     }
 
+    if (IsHigherVersion(version_hex) && trace_parent.size() < kTraceParentSize)
+    {
+      return SpanContext::GetInvalid();
+    }
+
+    if (!IsHigherVersion(version_hex) && trace_parent.size() != kTraceParentSize)
+    {
+      return SpanContext::GetInvalid();
+    }
```

Comment:
Seems we are now converting hex to binary thrice - line 156, 161, 166? This can be calculated once and reused.

Reply:
Yup, we can definitely do that. It would result in a nested-if though so might reduce the readability a little - but I'm ok to make this change.

Reply:
yes, it would be used in the hot-path for each request processed, so good to be optimized.

Reply:
I made some changes to accommodate this change. 

After taking out the code performing hex to binary conversions, I felt the private support functions could be replaced with simple if statements.